rpc/js/src/ClientChannel.ts

```diff
@@ -38,41 +52,26 @@ export class ClientChannel extends BaseChannel {
     dc.addEventListener('close', () => this.onConnectionTerminated());
   }
 
-  public transportFactory(): grpc.TransportFactory {
-    return (opts: grpc.TransportOptions) => {
-      return this.newStream(this.nextStreamID(), opts);
-    };
-  }
-
   private onConnectionTerminated() {
     // we may call this twice but we know closed will be true at this point.
     this.closeWithReason(new ConnectionClosedError('data channel closed'));
-    const err = new ConnectionClosedError('connection terminated');
     for (const streamId in this.streams) {
       const stream = this.streams[streamId]!;
-      stream.cs.closeWithRecvError(err);
+      stream.cs.closeWithRecvError();
     }
   }
 
   private onChannelMessage(event: MessageEvent<any>) {
-    let resp: Response;
-    try {
-      resp = Response.deserializeBinary(
-        new Uint8Array(event.data as ArrayBuffer)
-      );
-    } catch (e) {
-      console.error('error deserializing message', e);
-      return;
-    }
+    let resp = Response.fromBinary(new Uint8Array(event.data as ArrayBuffer));
```

Comment:
Do we need a runtime check or assert for this type cast to `as ArrayBuffer`? Or should `MessageEvent<any>` be defined more strictly?

Reply:
typed it more